parser_library/src/processing/preprocessors/cics_preprocessor.cpp

```diff
@@ -1004,6 +1011,27 @@ class cics_preprocessor final : public preprocessor
 
     static bool ignore_line(std::string_view s) { return s.empty() || s.front() == '*' || s.substr(0, 2) == ".*"; }
 
+    static bool is_ignored_line(std::string_view line, size_t line_len_chars)
+    {
+        if (ignore_line(line))
+            return true;
+
+        // apparently lines full of characters are ignored
+        if (line_len_chars == valid_cols && line.find(' ') == std::string_view::npos)
+            return true;
+
+        return false;
+    }
+
+    bool is_line_of_interest(std::string_view line)
+    {
+        static const std::regex line_of_interest("([^ ]*)[ ]+(START|CSECT|RSECT|DSECT|DFHEIENT|DFHEISTG|END)(?: .+)?");
+
+        return (std::regex_match(line.begin(), line.end(), m_matches_sv, line_of_interest)
+            && process_asm_statement(std::string_view(std::to_address(m_matches_sv[2].first), m_matches_sv[2].length()),
```

Comment:
the name `is_line_of_interest` implies that it just examines the line, but the `process_asm_statement` is actually doing stuff...

Reply:
Renamed